src/librustc/dep_graph/graph.rs

```diff
@@ -141,7 +141,20 @@ impl DepGraph {
         DepGraphQuery::new(&nodes[..], &edges[..])
     }
 
-    pub fn in_ignore<'graph>(&'graph self) -> Option<raii::IgnoreTask<'graph>> {
+    pub fn assert_ignored(&self)
+    {
+        if let Some(ref data) = self.data {
+            match data.current.borrow_mut().task_stack.last_mut() {
+                Some(&mut OpenTask::Ignore) |
+                Some(&mut OpenTask::EvalAlways { .. }) | None => {
```

Comment:
EvalAlways is conceptually not ignored. I'd remove this case.

Reply:
Also, you should be able to do non-mutable access here, right?

Reply:
I copied the ignored cases from `read_index`. Why is `EvalAlways` ignored there then?

Reply:
Because we artificial add a single edge to `DepNode::Krate` when the task finishes. This single edge will detect any change made to the crate or any upstream crates.